src/FlipMove.js

```diff
@@ -599,6 +608,25 @@ class FlipMove extends Component<void, ConvertedProps, FlipMoveState> {
     return this.state.children.find(child => getKey(child) === key);
   }
 
+  hasChildData(key: string): boolean {
+    return Object.prototype.hasOwnProperty.call(this.childrenData, key);
+  }
+
+  getChildData(key: string): NodeData {
+    return this.childrenData[key] || {};
+  }
+
+  putChildData(key: string, data: NodeData): void {
+    this.childrenData[key] = {
+      ...this.childrenData[key] || {},
```

Comment:
Also, this is a nit, but could we rename `putChildData` to `updateChildData`?

Could just be me, but a "put" to me feels like a hard replace, rather than a merge. "Update" indicates more clearly that "We're updating a field", not "we're replacing the old object with this new one"